pkg/testutils/mockmaps/metricsmap.go

```diff
@@ -38,3 +40,14 @@ func (m *MetricsMockMap) IterateWithCallback(cb metricsmap.IterateCallback) erro
 
 	return nil
 }
+
+// Delete removes the provided key from the map
+func (m *MetricsMockMap) Delete(key *metricsmap.Key) error {
+	for i, e := range m.Entries {
+		if e.Key == *key {
+			m.Entries = append(m.Entries[:i], m.Entries[i+1:]...)
```

Comment:
nit: could use [`slices.Delete`](https://pkg.go.dev/slices#Delete) here


```suggestion
			m.Entries = slices.Delete(m.Entries, i, i+1)
```